addons/sale/models/sale_order_line.py

```diff
@@ -850,6 +850,14 @@ def _compute_untaxed_amount_invoiced(self):
                         amount_invoiced -= invoice_line.currency_id._convert(invoice_line.price_subtotal, line.currency_id, line.company_id, invoice_date)
             line.untaxed_amount_invoiced = amount_invoiced
 
+    def _needs_remaining_calculation(self, inv_lines):
+        self.ensure_one()
```

Comment:
@lethuthao2403 you don't need `self` as a recordset, so no need to check `ensure_one()`